src/main/java/org/jboss/tm/usertx/TransactionListener.java

```diff
@@ -0,0 +1,16 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.Transaction;
+
+/**
+ * A listener callback for transaction related events
+ */
+public interface TransactionListener {
+    /**
+     * The callback for notifying registered listeners of a transaction related event
+     *
+     * @param transaction the transaction that the caller is registering interest in
+     * @param transactionEvent indication of what kind of change has occurred
+     */
+    void stateChange(Transaction transaction, TransactionEvent transactionEvent);
```

Comment:
Another note:
void stateChange(Transaction transaction, TransactionEvent transactionEvent)
To:
void onEvent(Transaction, TransactionEvent)

We might event none-state changes through the interface (such as 1PC commit occurred). Also, I think observer interfaces are often called back with "onXYZ".